src/lib/generate-key.c

```diff
@@ -254,7 +254,7 @@ static bool
 set_default_user_prefs(pgp_user_prefs_t *prefs)
 {
     if (!prefs->symm_algs) {
-        for (int i = 0; i < ARRAY_SIZE(DEFAULT_SYMMETRIC_ALGS); i++) {
+        for (unsigned i = 0; i < ARRAY_SIZE(DEFAULT_SYMMETRIC_ALGS); i++) {
```

Comment:
Technically it should be ``size_t`` as (in most cases) you compare to size_t not to ``unsigned ``. But all casts will work

Reply:
Agree, will change to size_t so things look more consistent.
Got to fun situation while playing with `-Wextra` - it doesn't allow `{0}` initializer for records while this is directly described in C standard.

Reply:
> Got to fun situation while playing with -Wextra - it doesn't allow {0} initializer for records while this is directly described in C standard.

Are you sure you don't just need `{{0}}` instead? IIRC the requirement is that you initialize the first field (and it cascades from there). If the first field is a struct, `{{0}}` or similar is needed.

Reply:
@dewyatt Hm, nope. For the following line:
`pgp_data_t OID = {0};` with the declaration:
```
typedef struct {
    size_t   len;
    uint8_t *contents;
    uint8_t  mmapped; /* contents need an munmap(2) */
} pgp_data_t;
```
Compiler tells:
```
error: missing field 'contents' initializer [-Werror,-Wmissing-field-initializers]
    pgp_data_t OID = {0};
```

Reply:
Ah, are you using clang? I think clang still has a bug there: https://bugs.llvm.org/show_bug.cgi?id=21689

Reply:
Yeah, as clang is default on macOS. Probably they will resolve the issue in a few more years :-D